### ISSUE TITLE
Expr::TrySuffix expression was not completely removed in desugar_value_def_suffixed 

### ISSUE THREAD

**Issue by bartonhammond** [roc-lang/basic-cli#297]
```
roc command.roc
An internal compiler expectation was broken.
This is definitely a compiler bug.
Please file an issue here: <https://github.com/roc-lang/roc/issues/new/choose>
a Expr::TrySuffix expression was not completely removed in desugar_value_def_suffixed
Location: crates/compiler/can/src/expr.rs:1254:40
```

If I do these steps it works fine

1. `jump-start.sh`
2. `roc build.roc`
3.  cd examples
4.  `roc command.roc`

If I update the `command.roc` first line to `app [main!] { pf: platform "https://github.com/roc-lang/basic-cli/releases/download/0.17.0/lZFLstMUCUvd5bjnnpYromZJXkQUrdhbva4xdBInicE.tar.br" }` I get this error

**Comment by Anton-4** [comment]
This will probably be fixed with the 0.18 release, which should be released soon, I just need to look into the CI failure [here](https://github.com/roc-lang/roc/pull/7410).

**Comment by Anton-4** [comment]
Yes, I've confirmed this is fixed with the [basic-cli 0.18.0 pre-release](https://github.com/roc-lang/basic-cli/releases/download/0.18.0/0APbwVN1_p1mJ96tXjaoiUCr8NBGamr8G8Ac_DrXR-o.tar.br).
